athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/saphana/SAPHANAMetadataHandler.java

```diff
@@ -0,0 +1,196 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.jdbc.saphana;
+
+import com.amazonaws.athena.connector.lambda.QueryStatusChecker;
+import com.amazonaws.athena.connector.lambda.data.Block;
+import com.amazonaws.athena.connector.lambda.data.BlockAllocator;
+import com.amazonaws.athena.connector.lambda.data.BlockWriter;
+import com.amazonaws.athena.connector.lambda.data.SchemaBuilder;
+import com.amazonaws.athena.connector.lambda.domain.Split;
+import com.amazonaws.athena.connector.lambda.domain.spill.SpillLocation;
+import com.amazonaws.athena.connector.lambda.metadata.GetSplitsRequest;
+import com.amazonaws.athena.connector.lambda.metadata.GetSplitsResponse;
+import com.amazonaws.athena.connector.lambda.metadata.GetTableLayoutRequest;
+import com.amazonaws.connectors.athena.jdbc.connection.DatabaseConnectionConfig;
+import com.amazonaws.connectors.athena.jdbc.connection.GenericJdbcConnectionFactory;
+import com.amazonaws.connectors.athena.jdbc.connection.JdbcConnectionFactory;
+import com.amazonaws.connectors.athena.jdbc.manager.JDBCUtil;
+import com.amazonaws.connectors.athena.jdbc.manager.JdbcMetadataHandler;
+import com.amazonaws.connectors.athena.jdbc.manager.PreparedStatementBuilder;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import org.apache.arrow.vector.complex.reader.FieldReader; 
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Handles metadata for SAPHANA. User must have access to `schemata`, `tables`, `columns`, `partitions` tables in
+ * information_schema.
+ */
+public class SAPHANAMetadataHandler
+        extends JdbcMetadataHandler
+{
+    static final Map<String, String> JDBC_PROPERTIES = ImmutableMap.of("databaseTerm", "SCHEMA");
+    static final String GET_PARTITIONS_QUERY = "SELECT DISTINCT part_id FROM SYS.TABLE_PARTITIONS WHERE TABLE_NAME = ? AND SCHEMA_NAME = ? ";
+    static final String BLOCK_PARTITION_COLUMN_NAME = "part_id";
+    static final String ALL_PARTITIONS = "*";
+    static final String PARTITION_COLUMN_NAME = "part_id";
+    private static final Logger LOGGER = LoggerFactory.getLogger(SAPHANAMetadataHandler.class);
+    private static final int MAX_SPLITS_PER_REQUEST = 1000_000;
```

Comment:
this seems really high, why 1M?

Reply:
Followed the practice from mySQL we have 1000_000 in there too.

Reply:
Thats fine, choose a name that uses CammelCase

Reply:
Fixed, please check